recipes/libcvd/all/conanfile.py

```diff
@@ -0,0 +1,180 @@
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/edrosten/libcvd"
+    topics = ("computer-vision",)
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_libjpeg": [False, "libjpeg", "libjpeg-turbo", "mozjpeg"],
+        "with_libpng": [True, False],
+        "with_libtiff": [True, False],
+        "with_ffmpeg": [True, False],
+        "with_libdc1394": [True, False],
+        "with_opengl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libjpeg": "libjpeg",
+        "with_libpng": True,
+        "with_libtiff": True,
+        # Build without video support by default
+        "with_ffmpeg": False,
+        "with_libdc1394": False,
+        "with_opengl": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            # OpenGL is always used on Windows
+            del self.options.with_opengl
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # https://github.com/edrosten/libcvd/blob/main/cmake/CVDFindAllDeps.cmake
+        if self.options.with_ffmpeg:
+            # FFMPEG v5.x+ are not supported
+            self.requires("ffmpeg/4.4.4", transitive_libs=True)
+        if self.options.with_libdc1394:
+            self.requires("libdc1394/2.2.7")
+            # FIXME: libidc1394 seems to be missing raw1394 dependency
+            # test_package fails with "undefined reference to `raw1394_new_handle'" etc
+        if self.options.with_libjpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/3.0.2")
+        elif self.options.with_libjpeg == "libjpeg":
+            self.requires("libjpeg/9e")
+        elif self.options.with_libjpeg == "mozjpeg":
+            self.requires("mozjpeg/4.1.5")
+        if self.options.with_libpng:
+            self.requires("libpng/[>=1.6 <2]")
+        if self.options.with_libtiff:
+            self.requires("libtiff/4.6.0")
+        if self.options.get_safe("with_opengl", True):
+            # https://github.com/edrosten/libcvd/blob/RELEASE_2_5_0/cvd/videodisplay.h#L18-L20
+            self.requires("opengl/system", transitive_headers=True, transitive_libs=True)
```

Comment:
Would it be possible to target `libglvnd` here on `Linux` and `FreeBSD` instead of `opengl/system`?

Reply:
That sounds good to me. My only general doubt is whether it could conflict with other packages still using `opengl/system`?
`libglvnd` does not set `provides = "opengl"`, but doesn't it effectively conflict with `opengl/system` nevertheless?

Reply:
You're totally correct about that, and I think we need to add that to the `libglvnd` recipe. However, I'd prefer to have less recipes to update in the future to use `libglvnd` on Linux and FreeBSD.